### PR TITLE
Add schema validation script/code

### PR DESCRIPTION
**Pull request by jskeet**
The intention is for this to be called from a CI test on
google-cloudevents, so that we can validate that schema changes will
work before merging them.

See https://github.com/googleapis/google-cloudevents/pull/52 for
sample data.

### CODE REVIEW

**Comment by jskeet** [comment]
Added @amanda-tarafa for the C# review and @grant for the overall validation plan.

File: src/Google.Events.ValidateSchema/Program.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            Console.WriteLine($"Failed: {failures}");
+            return failures == 0 ? 0 : 1;
+        }
+
+        /// <summary>
+        /// Validates that given test file can be parsed by the generated code.
+        /// </summary>
+        /// <param name="testRoot">Root directory of the test data</param>
+        /// <param name="file">Full name of the test file</param>
+        /// <returns>true of the test passes; false if it fails</returns>
```
Comment (amanda-tarafa) [inline src/Google.Events.ValidateSchema/Program.cs:74]: nit:
```suggestion
        /// <returns>true if the test passes; false if it fails</returns>
```

Comment (jskeet) [reply src/Google.Events.ValidateSchema/Program.cs:74]: Fixed, thanks.

**Comment by jskeet** [comment]
> It would be a lot better if we had the JSON representation next to the proto representation (next to the JSONSchema representation).

I very much disagree with this - the proto files are effectively "production code" and the JSON files are effectively "test code". I want to keep them separate just as I would for regular code.

> Another concern is .NET is not the most widely known/used language, so maintenance will be hard for serverless teams.

Every language should have an equivalent of this. That's why it's in google-cloudvents-dotnet, not google-cloudevents. The aim is that *every* language verifies that a schema change will be valid before it actually lands in google-cloudevents.

> Can we add README updates as to this feature?

There's a README update in the [PR in google-cloudevents](https://github.com/googleapis/google-cloudevents/pull/52); that's where it's most relevant.

**Pull request merged**